spark/src/main/scala/org/apache/spark/sql/delta/util/PartitionUtils.scala

```diff
@@ -528,46 +528,46 @@ private[delta] object PartitionUtils {
       timeZone: TimeZone,
       dateFormatter: DateFormatter,
       timestampFormatter: TimestampFormatter): Literal = {
-    val decimalTry = Try {
-      // `BigDecimal` conversion can fail when the `field` is not a form of number.
-      val bigDecimal = new JBigDecimal(raw)
-      // It reduces the cases for decimals by disallowing values having scale (eg. `1.1`).
-      require(bigDecimal.scale <= 0)
-      // `DecimalType` conversion can fail when
-      //   1. The precision is bigger than 38.
-      //   2. scale is bigger than precision.
-      Literal(bigDecimal)
-    }
+    if (typeInference) {
```

Comment:
Good refactor.

You could also make the `val decimalTry` a `def` instead so that it is only evaluated if necessary.

Reply:
Good suggestion, I like the def better it makes the diff easier on the eyes